smartapps/smartthings/samsung-smart-tv-connect.src/samsung-smart-tv-connect.groovy

```diff
@@ -0,0 +1,352 @@
+}
+
+def televisionDiscovery() {
+    state.permission = false
+    state.linkRefreshcount = -1
+    
+	def tvs = televisionsDiscovered()
+	int tvRefreshCount = !state.tvRefreshCount ? 0 : state.tvRefreshCount as int
+    
+    state.tvRefreshCount = tvRefreshCount + 1
+
+    def options = tvs ?: []
+    def numFound = options.size() ?: 0
+
+    if(!state.subscribe) {
+        subscribe(location, null, locationHandler, [filterEvents:false])
+        state.subscribe = true
+    }
+
+    if((tvRefreshCount % 5) == 0) {
+        discoverTVs()
+    }
+    
+    def unverified = getTVs().findAll { it?.value?.verified != true }
+    
+    if( (tvRefreshCount % 3) != 1 ) {
+    	sendHubIPReq(unverified)
+    }
+        
+    if( (tvRefreshCount % 3) != 2 ) {
+    	verifyTVs(unverified)
+    }
+    
+    return dynamicPage(name:"televisionDiscovery", title:"Discovery Started!", nextPage:"tvBtnPush", refreshInterval:refreshInterval, uninstall: true) {
+        section("Please wait while we discover your Samsung Smart TV. Discovery can take five minutes or more, so sit back and relax! Select your device below once discovered.") {
+            input "selectedTV", "enum", required:true, title:"Select Samsung Smart TV (${numFound} found)", multiple:false, options:options
+        }
+    }
+}
+
+def tvLinking() {
+	int linkRefreshcount = state.linkRefreshcount
+    
+    state.linkRefreshcount = linkRefreshcount + 1
+    
+	def nextPage = ""
+    def title = "Linking with your TV"
+	def paragraphText = "Press the 'Yes' button on a message, which is going to be shown on TV screen, to setup a link."
+    
+    if (state.permission)
+    {
+    	nextPage = "tvName"
+		title = "Success! - click 'Next'"
+		paragraphText = "Linking to your hub was a success! Please click 'Next'!"
+    }
+    else if( !state.permission && ((linkRefreshcount % 5) == 0) )
+    {
+    	sendDeveloperReq()
+    }
+    
+    return dynamicPage(name:"tvBtnPush", title:title, nextPage:nextPage, refreshInterval:refreshInterval) {
+    	section("Button Press") {
+        	paragraph """${paragraphText}"""
+        }
+    }
+}
+
+def tvName() {
+    return dynamicPage(name:"tvName", title:"Set Name!", install:true) {
+		section{
+			input "TVname", "text", title:"Give your Samsung Smart TV a name", required:true
+		}
+    }
+}
+
+private discoverTVs() {
+    sendHubCommand(new physicalgraph.device.HubAction("lan discovery urn:samsung.com:device:SmartThingsTvDevice:1", physicalgraph.device.Protocol.LAN))
+}
+
+private sendHubIPReq(devices) {
+    devices.each {
+       	def hubID = it.value.hubID
+       	def vHub = location.hubs.find{it.id == hubID}
+        def body = "{\"request\" : \"set\", \"type\" : \"hubIP\", \"value\" : \"${vHub.localIP + ":" + vHub.localSrvPortTCP}\"}"
+        sendHubCommand(new physicalgraph.device.HubAction([
+        method: "POST",
+        path: "/smartthings/hub",
+        headers: [
+            HOST: ipAddressFromDni(it.value.ip + ":" + it.value.port),
```

Comment:
This value should not be populated from the DNI, but from IP and Port values associated with the TV.